architecture/2024-11-authz-feature-spec.md

```diff
@@ -0,0 +1,390 @@
+| Environment Administrator    | env-admin    | Applications.Core/environments                       | *              |
+|                              |              | Applications.Core/secretStores                       | *              |
+|                              |              | System.AWS/credentials                               | *              |
+|                              |              | System.Azure/credentials                             | *              |
+|                              |              | System.Kubernetes/credentials (future functionality) | *              |
+| Resource Type Administrator  | type-admin   | System.Resources/resourceproviders                   | *              |
+| Recipe Administrator         | recipe-admin | Only the recipe property of environments             | *              |
+
+#### User story 1 – Assigning the Radius Administrator role
+
+> [!NOTE]
+>
+> In a new Radius installation, there is no identity provider (IdP) configured. Radius does not have a built-in identity system, therefore the Radius administrator must configure an IdP before Radius will recognize and authorize the user. Roles can be assigned prior to IdP configuration, but without an IdP there is no effect. 
+
+As a platform engineer, I just installed Radius using my Kubernetes cluster credentials. I want to access Radius via my typical user credentials from my IdP.
+
+**User Experience**
+
+```bash
+# Assign the Radius Administrator role to a user from the IdP
+rad role assignment create \
+  --assignee me@my-company.net \
+  --role radius-admin \
+  --scope /planes/radius/*
+```
+
+**Result**
+
+1. `me@my-company.net` is added to the out-of-the-box role `radius-admin` 
+2. The user can now authenticate to Radius using the IdP and continue administering Radius with the users standard credentials
+
+**Exceptions**
+
+The operation fails and informs the user interactively if:
+
+* The current user is not a member of the `cluster-admin` RoleBinding in the Kubernetes cluster
+
+#### User story 2 – Assigning the Developer role to the default application resource group
+
+As a platform engineer, I need to assign the Developer role to a developer for the first time. There are no applications created in Radius yet and I am not familiar with resource groups yet.
+
+ **User Experience**
+
+When Radius is installed, two resource groups are created:
+
+1. `default-application` – When a user deploys an application without specifying a resource group, it is deployed to this resource group.
+2. `default-environment` – When a user creates an environment without specifying a resource group, it is deployed to this resource group.
+
+> [!NOTE]
+>
+> The default resource group is split to encourage best practices for actual Radius use. Splitting resource groups introduces the concept of resource groups and Radius' authorization model.
```

Comment:
Not sure I understand fully why the split at the resourcegroup.

Reply:
The pattern we are encouraging is applications are held in one set of resource groups while environments are held in a separate set of resource groups.